src/widgetApp/mainwindow.cpp

```diff
@@ -1,16 +1,311 @@
 #include "mainwindow.h"
 #include "ui_mainwindow.h"
 
+#include <QSpinBox>
+#include <QLabel>
+#include <QtCore>
+
+#include "IPv4parser.h"
+#include "SubnetsCalculatorV4.h"
+#include "IPstructs.h"
+
+using namespace core;
+
 namespace widgetApp {
+
     MainWindow::MainWindow(QWidget *parent)
         : QMainWindow(parent)
         , ui(new Ui::MainWindow)
     {
         ui->setupUi(this);
+
+        addressWidget = ui->Address4Widget;
+
+        subnetScrollContent = ui->subnetScroll;
+        subnetScrollContent->setLayout(subnetsPanelLayout);
+
+        infoScrollContent = ui->infoScroll;
+        infoScrollContent ->setLayout(infoPanelLayout);
+
+        subnetGraphContent = ui->graphScroll;
+
+        subnetCountBox = ui->hostNumberSpinBox;
+
+        binaryAddressWidget = ui->BinaryAddress4Widget;
+
+        maskWidget = ui->Mask4Widget;
+
+        binaryMaskWidget = ui->BinaryMask4Widget;
     }
 
     MainWindow::~MainWindow()
     {
         delete ui;
     }
 }
+
+void widgetApp::MainWindow::resetData()
+{
+    mainNetwork = Networkv4();
+    calculator = SubnetsCalculatorV4();
+
+    subnetCount = 0;
+    isHorizontal = false;
+}
+
+void widgetApp::MainWindow::on_drawButton_clicked()
+{
+    drawNetworkGraph(isHorizontal);
+}
+
+void widgetApp::MainWindow::on_radioButton_clicked(bool checked)
+{
+    isHorizontal = (checked) ? true : false;
+    drawNetworkGraph(isHorizontal);
+}
+
+void widgetApp::MainWindow::on_CalculateButton_clicked()
+{
+     mainNetwork.Ip = parser.ipFromString(
+                 takeStringFromInputFields(addressWidget));
+
+     mainNetwork.NetMask = parser.ipMaskFromString(
+                 takeStringFromInputFields(maskWidget));
+
+     displayInputInBinary(mainNetwork.Ip.get()->asStringBin(),
+                          binaryAddressWidget);
+
+     displayInputInBinary(mainNetwork.NetMask.get()->asStringBin(),
+                          binaryMaskWidget);
```

Comment:
get - jest metodą shared_ptr, nie ma potrzeby pobierania wskaźnika.
Ip i Net mask są shared_ptr, więc wystarczy je zdereferować, tak samo jak w `line 289`
```suggestion
     displayInputInBinary(mainNetwork.Ip->asStringBin(),
                          binaryAddressWidget);

     displayInputInBinary(mainNetwork.NetMask->asStringBin(),
                          binaryMaskWidget);
```